beetsplug/parentwork.py

```diff
@@ -0,0 +1,225 @@
+# -*- coding: utf-8 -*-
+# This file is part of beets.
+# Copyright 2017, Dorian Soergel.
+#
+# Permission is hereby granted, free of charge, to any person obtaining
+# a copy of this software and associated documentation files (the
+# "Software"), to deal in the Software without restriction, including
+# without limitation the rights to use, copy, modify, merge, publish,
+# distribute, sublicense, and/or sell copies of the Software, and to
+# permit persons to whom the Software is furnished to do so, subject to
+# the following conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+
+"""Gets work title, disambiguation, parent work and its disambiguation,
+composer, composer sort name and performers
+"""
+
+from __future__ import division, absolute_import, print_function
+
+from beets import ui
+from beets.plugins import BeetsPlugin
+
+import musicbrainzngs
+
+
+def work_father(work_id, work_date=None):
+    """ This function finds the id of the father work given its id"""
+    work_info = musicbrainzngs.get_work_by_id(work_id,
+                                              includes=["work-rels",
+                                                        "artist-rels"])
+    if 'artist-relation-list' in work_info['work'] and work_date is None:
+        for artist in work_info['work']['artist-relation-list']:
+            if artist['type'] == 'composer':
+                if 'end' in artist.keys():
+                    work_date = artist['end']
+
+    if 'work-relation-list' in work_info['work']:
+        for work_father in work_info['work']['work-relation-list']:
+            if work_father['type'] == 'parts' \
+                    and work_father.get('direction') == 'backward':
+                father_id = work_father['work']['id']
+                return father_id, work_date
+    return None, work_date
+
+
+def work_parent(work_id):
+    """This function finds the parentwork id of a work given its id. """
+    work_date = None
+    while True:
+        (new_work_id, work_date) = work_father(work_id, work_date)
+        if not new_work_id:
+            return work_id, work_date
+            break
+        work_id = new_work_id
+    return work_id, work_date
+
+
+def find_parentwork(work_id):
+    """This function gives the work relationships (dict) of a parent_work
+    given the id of the work"""
+    parent_id, work_date = work_parent(work_id)
+    work_info = musicbrainzngs.get_work_by_id(parent_id,
+                                              includes=["artist-rels"])
+    return work_info, work_date
+
+
+class ParentWorkPlugin(BeetsPlugin):
+    def __init__(self):
+        super(ParentWorkPlugin, self).__init__()
+
+        self.config.add({
+            'auto': False,
+            'force': False,
+        })
+
+        self._command = ui.Subcommand(
+            'parentwork',
+            help=u'Fetches parent works, composers and dates')
```

Comment:
These `help` strings are typically lower case and in the imperative voice, i.e., "fetch parent works…"